documentation/logs.md

```diff
@@ -0,0 +1,152 @@
+# Logs
+
+Bolt supports multiple log levels. You can configure the log level from the CLI,
+or in a project configuration file. Supported logging levels, in order from most
+to least information logged, are `trace`, `debug`, `info`, `warn`, and `error`.
+
+By default, Bolt logs a debug-level log to a `bolt-debug.log` file in the root
+of your project directory. You can also configure custom logs from your project
+configuration file.
+
+## The `bolt-debug.log` file
+ 
+Each time you run a Bolt command, Bolt prints a debug-level log to a
+`bolt-debug.log` file in the root of your project directory.
+
+You can disable the log file by specifying the following in your
+`bolt-project.yaml` file:
+
+```yaml
+log:
+  bolt-debug.log: disable
+```
+
+## Setting log level
+
+You can set Bolt's log level from the CLI or use a
+configuration file.
+
+### Setting log level on the CLI
+
+To set the log level from the CLI, use the `--log-level` option along with the
+desired level. Available log levels are `trace`, `debug`, `info`, `warn`, and
+`error`. For example:
+
+```console
+bolt command run whoami -t target1 --log-level trace
+```
+
+### Setting log level in a configuration file
+
+To set the log level for the console, add a `log` map with a `console` mapping
+to your [project configuration file](configuring_bolt.md#project-level-configuration).
+
+Use `level` to set the log level. Available log levels are `trace`, `debug`,
+`info`, `warn`, and `error`. For example:
+
+```yaml
+# bolt-project.yaml
+name: lotsalogs
+
+log:
+  console:
+    level: debug
+```
+
+If you want to write a log to a file, add a map with the file destination to the
+log mapping. For example, the following `bolt-project.yaml` file writes `info`
+and `trace` level logs to a `logs` directory in the Bolt project:
+
+```yaml
+# bolt-project.yaml
+name: lotsalogs
+
+log:
+  "logs/trace.log":
+    append: false
+    level: trace
+  "logs/info.log":
+    append: false
+    level: info
+```
+
+Set the `append` key to `false` if you want to overwrite the file on each run,
+or `true` if you want to append to the existing file.
+ 
+> **Note**: The directory you set for logging must exist, and Bolt must have
+> permission to write to it.
+
+## Log levels
+
+### `trace`
+
+Trace logs contain the most detailed information for actions run on individual
+targets. For example, the logs include each command run on a target, each step
+of compiling a manifest for an apply block, and information about establishing
+connections with targets.
```

Comment:
I'm not sure readers will find this distinction useful, but the team thinks of trace as "why Bolt is doing what it's doing" while debug is just "what Bolt's doing"